libut/Makefile

```diff
@@ -1,33 +1,41 @@
-OBJS=libut.a
-all: $(OBJS) 
+SHELL=/usr/sh
+VERSION=2.0.1
+MAJOR=2
+SOURCES=$(shell echo src/*.c)
+HEADERS=$(shell echo include/*.h)
+OBJS=$(SOURCES:.c=.o)
+all: libut.a libut.so.$(VERSION)
 INCDIR=./include
+LIBDIR=/usr/lib
 CFLAGS+=-I$(INCDIR)
-CFLAGS+=-Wall -Wextra
+CFLAGS+=-Wall -Wextra -Werror -fPIC
 CFLAGS+=-g
+CFLAGS+=$(OPTFLAGS)
 
-libut.a: libut.o utvector.o utmm.o ringbuf.o
+libut.a: $(OBJS)
 	ar r $@ $^
 
-libut.o: src/libut.c $(INCDIR)/libut.h
-	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
-
-utvector.o: src/utvector.c $(INCDIR)/utvector.h
-	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
-
-utmm.o: src/utmm.c $(INCDIR)/utmm.h
-	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
-
-ringbuf.o: src/ringbuf.c $(INCDIR)/ringbuf.h
-	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
+libut.so.$(VERSION): $(OBJS)
+	$(CC) $(CFLAGS) -shared -Wl,-soname,libut.so.$(MAJOR) $^ -o $@
 
 .PHONY: clean tests install
 
 clean:
-	rm -f $(OBJS) *.o
+	rm -f *.o *.a *.so*
 	make -C tests clean
 
 tests: libut.a
 	make -C tests 
 
-install: libut.a
-	cp $< /usr/local/lib
+install: libut.a libut.so.$(VERSION)
+	mkdir -p $(DESTDIR)$(LIBDIR)
+	mkdir -p $(DESTDIR)$(LIBDIR)/pkgconfig
+	mkdir -p $(DESTDIR)/usr/include/uthash
+	cp libut.a $(DESTDIR)$(LIBDIR)
+	cp libut.so.$(VERSION) $(DESTDIR)$(LIBDIR)
+	ln -sf $(LIBDIR)/libut.so.$(VERSION) $(DESTDIR)$(LIBDIR)/libut.so.$(MAJOR)
+	ln -sf $(LIBDIR)/libut.so.$(VERSION) $(DESTDIR)$(LIBDIR)/libut.so
+	cp ./include/*.h $(DESTDIR)/usr/include/uthash
```

Comment:
Right. it needs extraordinary evidence. I think: since we have many ut*.h header files, not just one or two. then why not gather them into a directory/namespace and leave an universal header outside in /usr/include? that looks just nice...and what glib did...if everyone decide to put many headers inside /usr/include directly, finally there'll be conflicts in names. eg if someone develops a "universal timer", there's a great chance that `utvector.h` looks good to him too. 

In openSUSE, we branch it, fix it, then submit it. That you found is the central portal. Just like you can't find the latest "fix" in kernel main branch...I didn't paste my branch because it will be gone after my SR is accepted...https://build.opensuse.org/package/show/home:MargueriteSu:branches:devel:libraries:c_c++/uthash